kernel/subtyping.ml

```diff
@@ -246,18 +246,23 @@ let check_constant (cst, ustate) trace env l info1 cb2 subst1 subst2 =
       (* Now we check the bodies:
          - A transparent constant can only be implemented by a compatible
            transparent constant.
+         - A primitive cannot be implemented.
+           (We could try to allow implementing with the same primitive,
+            but for some reason we get cb1.const_body = Def,
```

Comment:
Presumably this is "NB: cb1 might have been strengthened and appear as transparent.", and the way to handle it would be to have conv_check check the constant is convertible with the primitive?  But maybe we want to forbid it anyway the same way we seem to forbid
```coq
Module A. Axiom X : Set. End A.
Module Type AT. Definition X : Set := A.X. End AT.
Module F (Av : AT). Definition Y := Av.X. End F.
Module G := F A.
```
(This does get rejected, right?  I'm not in front of a computer right now so can't check.)